crates/install-wheel-rs/src/linker.rs

```diff
@@ -351,91 +363,62 @@ enum Attempt {
     UseCopyFallback,
 }
 
-/// Recursively clone the contents of `from` into `to`.
-fn clone_recursive(
+/// Recursively clone the contents of `from` into `to` with fallback to copy if not supported.
+fn clone_or_copy_recursive(
     site_packages: &Path,
     wheel: &Path,
     locks: &Locks,
     entry: &DirEntry,
+    reflink_directories: bool,
     attempt: &mut Attempt,
 ) -> Result<(), Error> {
     // Determine the existing and destination paths.
     let from = entry.path();
     let to = site_packages.join(from.strip_prefix(wheel).unwrap());
 
-    debug!("Cloning {} to {}", from.display(), to.display());
-
-    if cfg!(windows) && from.is_dir() {
-        // On Windows, reflinking directories is not supported, so we copy each file instead.
-        fs::create_dir_all(&to)?;
-        for entry in fs::read_dir(from)? {
-            clone_recursive(site_packages, wheel, locks, &entry?, attempt)?;
-        }
-        return Ok(());
-    }
```

Comment:
Why would it not be sufficient to extend this to Linux?

Reply:
It is. This is the first commit in the PR, actually.

I then tried to decouple a bit this code for the next person who needs to touch it. I found it a bit difficult to reason about because it was handling 2 separate fallback logic, plus the windows and linux special case in `clone_recursive`.

I do think this makes it easier to follow, but I understand this is somewhat subjective. If you prefer, I can stick with the minimal change that fixes the bug.

Reply:
While I definitely appreciate the intent, I would prefer to use the minimal change, since this function is _so_ critical and making large changes to it requires extensive testing.

Reply:
Ok. I updated the branch with a minimal change to fix the bug.

Reply:
Thanks. This I can easily and confidently merge.